tests/PHPUnit/Integration/PagesTest.php

```diff
@@ -0,0 +1,149 @@
+ * along with this program.
+ * If not, see <http://www.pdfparser.org/sites/default/LICENSE.txt>.
+ */
+
+namespace PHPUnitTests\Integration;
+
+use PHPUnitTests\TestCase;
+use Smalot\PdfParser\Document;
+use Smalot\PdfParser\Element\ElementArray;
+use Smalot\PdfParser\Font;
+use Smalot\PdfParser\Header;
+use Smalot\PdfParser\Page;
+use Smalot\PdfParser\Pages;
+
+/**
+ * @internal only for test purposes
+ */
+class PagesDummy extends Pages
+{
+    /**
+     * The purpose of this function is to bypass the tedious
+     * work to setup instances which lead to a valid $fonts variable.
+     *
+     * @param array<\Smalot\PdfParser\Font> $fonts
+     *
+     * @return void
+     */
+    public function setFonts($fonts)
+    {
+        $this->fonts = $fonts;
+    }
+}
+
+class PagesTest extends TestCase
+{
+    /**
+     * If fonts are not stored in Page instances but in the Pages instance.
+     *
+     *      Pages
+     *        |   `--- fonts = Font[]           <=== will be used to override fonts in Page1 ...
+     *        |
+     *        |
+     *        `--+ Page1
+     *        |         `--- fonts = null       <=== Will be overwritten with the content of Pages.fonts
+     *        `--+ ...
+     *
+     * @see https://github.com/smalot/pdfparser/pull/698
+     */
+    public function testPullRequest698NoFontsSet(): void
+    {
+        $document = $this->createMock(Document::class);
+
+        // Create a Page mock and tell PHPUnit that its setFonts has to be called once
+        // otherwise an error is raised
+        $page1 = $this->createMock(Page::class);
+        $page1->expects($this->once())->method('setFonts');
+
+        // setup header
+        $header = new Header([
+            'Kids' => new ElementArray([
+                $page1,
+            ]),
+        ], $document);
+
+        $font1 = $this->createMock(Font::class);
+
+        // Preset fonts variable so we don't have to prepare all the
+        // prerequisites manually (like creating a Ressources instance
+        // with Font instances, see Pages::setupFonts())
+        $pages = new PagesDummy($document, $header);
+        $pages->setFonts([$font1]);
+
+        // We expect setFonts is called on $page1, therefore no assertion here
+        $pages->getPages(true);
+    }
+
+    /**
+     * Dont override fonts list in a Page instance, if available.
+     *
+     *      Pages
+     *        |   `--- fonts = Font[]           <=== Has to be ignored because fonts in Page1 is set
+     *        |
+     *        |
+     *        `--+ Page1
+     *        |         `--- fonts = Font[]     <=== must not be overwritten
+     *        `--+ ...
+     *
+     * @see https://github.com/smalot/pdfparser/pull/698
+     */
+    public function testPullRequest698DontOverride(): void
+    {
+        $document = $this->createMock(Document::class);
+
+        // create a Page mock and tell PHPUnit that its setFonts has to be called once
+        // otherwise an error is raised
```

Comment:
It seems this comment is duplicated from the function above, but it isn't what we're actually testing here.